packages/client/hmi-client/src/components/model-template/tera-model-template-editor.vue

```diff
@@ -140,26 +161,42 @@ let currentPortPosition: Position = { x: 0, y: 0 };
 let isMouseOverCanvas = false;
 let canvasTransform = { x: 0, y: 0, k: 1 };
 let isMouseOverPort = false;
+let junctionIdForNewEdge: number | null = null;
 
-const modelTemplates = ref<any[]>([]); // ([{ id: 1, name: 'Template name', x: 300, y: 40 }]);
-const edges = ref<ModelTemplateEdge[]>([]);
+const modelTemplates = ref<any[]>([]);
+const junctions = ref<ModelTemplateJunction[]>([]);
 
 const newModelTemplate = ref();
 const newEdge = ref();
 const isCreatingNewEdge = computed(
 	() => newEdge.value && newEdge.value.points && newEdge.value.points.length === 2
 );
 
+function collisionFn(p: Position) {
+	const buffer = 50;
+	for (let i = 0; i < modelTemplates.value.length; i++) {
+		const checkingNode = modelTemplates.value[i].metadata.templateCard;
+		// FIXME: Thi is  a hack to get around hierarhical geometries, will need to
+		// relax this guard.
+		// if (node.nodes && node.nodes.length > 0) continue;
+		if (p.x >= checkingNode.x - buffer && p.x <= checkingNode.x + checkingNode.width + buffer) {
+			if (p.y >= checkingNode.y - buffer && p.y <= checkingNode.y + checkingNode.height + buffer) {
+				return true;
+			}
+		}
+	}
+	return false;
+}
+
 function interpolatePointsForCurve(a: Position, b: Position): Position[] {
-	const controlXOffset = 0;
-	return [a, { x: a.x + controlXOffset, y: a.y }, { x: b.x - controlXOffset, y: b.y }, b];
+	return getAStarPath(a, b, collisionFn);
 }
```

Comment:
Most of the performance will be gained by not calculating the AStar-path in rendering.

If we stick with AStar,
- You want to calculate out `edge.points` on node-dragging events, and then 
- Here you'd just return the calculated `edges.points` ... e.g. you can remove this function


pseudo code:
```
const updatePosition = ({ x, y }, node: any) => {
  .... stuff ...
  const edges_going_in = get_edges_that_are_input(node)
  const edges_going_out = get_edges_that_are_output(node)
  
  edges_going_in.forEach(e => {
    e.points = getAStarPath(_.first(e.points), getPortPosition(node, e))
  })
  edges_going_out.forEach(e => {
    e.points = getAStarPath(_.last( getPortPosition(node, e), _.last(e.points))
  })
}
```